### PR TITLE
Search on query type

### PR DESCRIPTION
**Pull request by Winwardo**
Ready for comment, not for merge yet - might be missing an optimisation for .popups

### CODE REVIEW

**Comment by Winwardo** [comment]
Ready for comments AND merge

**Comment by VasilyShelkov** [comment]
Cool, will take a look when I get back from my interview... offices are banging in pie mapping... based in a London tech hub X) very cool

File: src/server/tweetFinder.js [hunk 1]
```diff
@@ -73,30 +73,86 @@ const resultsToPresentableOutput = (results) => (
...
+};
+
+const searchByHashtag = (keyword) => {
+  const tweetSelection = makeTweetQuerySelectingFrom(
+    'TRAVERSE in(\'HAS_HASHTAG\') FROM (SELECT FROM hashtag WHERE content LUCENE :query)'
+  );
+  return db.query(tweetSelection, { params: { query: `${keyword}~`, limit: 300 } });
+};
+
+const makeTweetQuerySelectingFrom = (from) => (
```
Comment (VasilyShelkov) [inline src/server/tweetFinder.js:131]: hmmm why is from highlighting here ? it's not a javascript or node keyword is it :S ?

File: src/client/search/searchActions.js [hunk 2]
```diff
@@ -35,9 +35,10 @@ export const RECEIVE_FEED_RESULTS = 'RECEIVE_FEED_RESULTS';
 export const searchApiForFeed = (searchTerms) =>
 (dispatch) => (
   newPromiseChain()
-  .then(() => (fetchPost('/search', searchTerms)))
-  .then(response => (response.json()))
-  .then(json => dispatch({ type: RECEIVE_FEED_RESULTS, data: json }))
+    .then(() => NProgress.start())
```
Comment (VasilyShelkov) [inline src/client/search/searchActions.js:38]: I'm confused, what's this Nprogress actually doing here :S ? I can't see it being used anywhere or where NProgress even comes from

**Comment by VasilyShelkov** [comment]
@Winwardo, overall happy with this, just if you could answer the 2 minor comments X)

File: src/server/tweetFinder.js (hunk 1 above)
Comment (Winwardo) [inline src/server/tweetFinder.js:131]: Not according to any guides I've seen, nor is it highlighted in WebStorm. I think it's safe, but can change it if we're not sure?

File: src/client/search/searchActions.js (hunk 2 above)
Comment (Winwardo) [inline src/client/search/searchActions.js:38]: It's magic.
Happy to remove NProgress entirely once we've put in proper semantic spinners, but it's a global object with 4 methods on it http://ricostacruz.com/nprogress/

Comment (VasilyShelkov) [inline src/client/search/searchActions.js:38]: sure, I'll trust you X)

**Pull request merged**